tests/io/test_file_info.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from pathlib import Path
+import pytest
+
+from sedpack.io.file_info import DirectoryGenerator
+
+
+def test_multilevel() -> None:
+    levels: int = 3
+    max_branching: int = 2
+    name_length: int = 3
+
+    generator = DirectoryGenerator(
+        levels=levels,
+        max_branching=max_branching,
+        name_length=name_length,
+    )
+
+    seen_names: set[str] = set()
+
+    # No exception in the top level
+    for _ in range((max_branching**levels) + 10):
+        n = generator.get_path()
+        assert n.count("/") == levels - 1
+        p = Path(n)
+        assert len(p.parts) == levels
+
+        # Last part is long, previous are at most name_length
+        assert len(p.parts[-1]) >= 32
+        assert all(len(part) == name_length for part in p.parts[:-1])
+
+        # Enforce format: name/name/name/long_name
+        for l in range(1, levels):
```

Comment:
Isn't that a duplicate from the assert above?

Reply:
[obsolete, generating Path]

This was useful later for checking number of subdirectories.